### PR TITLE
feat: exclusive relayer

### PR DESCRIPTION
**Pull request by gsteenkamp89**
## Exclusive relayer epic branch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,6 +24,7 @@
 src/data/fill-times-preset.json
 src/data/exclusive-relayer-configs.json
 src/data/exclusive-relayer-weights.json
+src/data/exclusivity-strategy.json
 src/data/rpc-providers.json
```
Comment (pxrl) [inline .gitignore:28]: ooc, should we just blanket ignore `src/data/*.json` ?

File: scripts/pre-build.ts [hunk 2]
```diff
@@ -80,6 +81,17 @@ const remoteConfigs = {
     },
     localFilePath: "src/data/exclusive-relayer-weights.json",
   },
+  [remoteConfigTypes.EXCLUSIVITY_STRATEGY]: {
+    fetchFn: () =>
+      fetchExclusivityConfig(
+        // TODO: Update below value to the correct remote URL and file path when the remote config is ready.
```
Comment (pxrl) [inline scripts/pre-build.ts:87]: Nuke

```suggestion
```

File: .gitignore (hunk 1 above)
Comment (dohaki) [reply .gitignore:28]: In the near future, yes. But currently we also have some files in `src/data` that can not be ignored like `routes_*.json`

File: api/batch-account-balance.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    logger.debug({
+      at: "BatchAccountBalance",
+      message: "Response data",
+      responseJson: data,
+    });
+
+    // Set the caching headers that will be used by the CDN.
+    response.setHeader(
+      "Cache-Control",
+      "s-maxage=150, stale-while-revalidate=150"
```
Comment (dohaki) [inline api/batch-account-balance.ts:102]: Should we decrease those to 60 seconds or even 30?

**Review by dohaki — APPROVED** [review]
Looking good to me 👍

File: .gitignore (hunk 1 above)
Comment (dohaki) [reply .gitignore:28]: Maybe after we move the routes file also into the config repo?

Comment (pxrl) [reply .gitignore:28]: SGTM!

File: api/batch-account-balance.ts (hunk 3 above)
Comment (pxrl) [reply api/batch-account-balance.ts:102]: Yeah 30s sounds good to me.

Comment (pxrl) [inline api/batch-account-balance.ts:102]:
```suggestion
      "s-maxage=30, stale-while-revalidate=30"
```

**Pull request merged**